master/internal/db/postgres_jobs_intg_test.go

```diff
@@ -0,0 +1,168 @@
+
+		// retrieve job and test for equality
+		recvJob, err := db.JobByID(sendJob.JobID)
+		require.NoError(t, err)
+		assert.Equal(t, sendJob.JobID, recvJob.JobID)
+		assert.Equal(t, sendJob.JobType, recvJob.JobType)
+		assert.Equal(t, sendJob.OwnerID, recvJob.OwnerID)
+		assert.Equal(t, decimal.NewFromInt(10).Equal(recvJob.QPos), true)
+	})
+
+	t.Run("retrieve non-existent job", func(t *testing.T) {
+		// retrieve job and test for equality
+		recvJob, err := db.JobByID(model.NewJobID())
+		require.Error(t, err)
+		require.Nil(t, recvJob)
+	})
+}
+
+func TestUpdateJobPosition(t *testing.T) {
+	db := setupDBForTest(t)
+
+	t.Run("update position", func(t *testing.T) {
+		// create and send job
+		origPos := decimal.NewFromInt(10)
+		newPos := decimal.NewFromInt(5)
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    origPos,
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// update job position
+		err = db.UpdateJobPosition(sendJob.JobID, newPos)
+		require.NoError(t, err)
+
+		// retrieve job and confirm pos update
+		recvJob, err := db.JobByID(sendJob.JobID)
+		require.NoError(t, err)
+		assert.Equal(t, newPos.Equal(recvJob.QPos), true)
+	})
+
+	t.Run("update position - negative value", func(t *testing.T) {
+		// create and send job
+		origPos := decimal.NewFromInt(10)
+		newPos := decimal.NewFromInt(-5)
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    origPos,
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// update job position
+		err = db.UpdateJobPosition(sendJob.JobID, newPos)
+		require.NoError(t, err)
+
+		// retrieve job and confirm pos update
+		recvJob, err := db.JobByID(sendJob.JobID)
+		require.NoError(t, err)
+		assert.Equal(t, newPos.Equal(recvJob.QPos), true)
+	})
+
+	t.Run("update position - empty ID", func(t *testing.T) {
+		// create and send job
+		origPos := decimal.NewFromInt(10)
+		newPos := decimal.NewFromInt(5)
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    origPos,
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// update job position
+		err = db.UpdateJobPosition(model.JobID(""), newPos)
+		require.Error(t, err)
+
+		// retrieve job and ensure queue pos not updated
+		recvJob, err := db.JobByID(sendJob.JobID)
+		require.NoError(t, err)
+		assert.Equal(t, origPos.Equal(recvJob.QPos), true)
+	})
+
+	t.Run("update position - ID does not exist", func(t *testing.T) {
+		// create and send job
+		origPos := decimal.NewFromInt(10)
+		newPos := decimal.NewFromInt(5)
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    origPos,
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// update job position for a job that doesn't exist
+		err = db.UpdateJobPosition(model.NewJobID(), newPos)
+		require.NoError(t, err)
+	})
+}
```

Comment:
blocker: since this test has a lot of repeated code, I recommend setting this test up with test cases and then looping through them, see https://dave.cheney.net/2019/05/07/prefer-table-driven-tests or search for `cases := []struct {` in the repo.

Reply:
I appreciate the desire to limit repeated code! I also see your point - even though I disagree. 

- the tests aren't exact copies of each other; I think it will add complexity to try and fit them all in the same table test. Basically, there's two types of variations: position value and job ID. 
- you can't run a single test case with table tests (as far as I can tell); since we prefer `require` over `assert`, this means the test stops at the first test case that fails. With the existing code structure we can run individual cases or run `TestUpdateJobPosition` and get info on all failing cases

Repeated code is usually more tolerated in test code because the purpose is a little different. It's more important to be usable and to easily understand what a test is doing. 

I think a nice compromise might be to extract some of the repeated code into a helper function. How does that sound to you?

Reply:
I think that's a good idea